mmengine/utils/package_utils.py

```diff
@@ -9,28 +12,19 @@ def is_installed(package: str) -> bool:
     Args:
         package (str): Name of package to be checked.
     """
-    # When executing `import mmengine.runner`,
-    # pkg_resources will be imported and it takes too much time.
-    # Therefore, import it in function scope to save time.
     import importlib.util
 
-    import pkg_resources  # type: ignore
-    from pkg_resources import get_distribution
-
-    # refresh the pkg_resources
-    # more datails at https://github.com/pypa/setuptools/issues/373
-    importlib.reload(pkg_resources)
+    # First check if it's an importable module
+    spec = importlib.util.find_spec(package)
+    if spec is not None and spec.origin is not None:
+        return True
+    
+    # If not found as module, check if it's a distribution package
     try:
-        get_distribution(package)
+        distribution(package)
         return True
-    except pkg_resources.DistributionNotFound:
-        spec = importlib.util.find_spec(package)
-        if spec is None:
-            return False
-        elif spec.origin is not None:
-            return True
-        else:
-            return False
+    except PackageNotFoundError:
```

Comment:
Removing `importlib.util.find_spec(package)` will introduce BC. For example, the previous `is_installed("PIL")` or `is_installed("Pillow")` could return `True` since `PIL` is a package name and `Pillow` is the distribution name.